src/EA.Weee.RequestHandlers.Tests.Unit/Organisations/OrganisationByIdHandlerTests.cs

```diff
@@ -379,16 +377,143 @@ public async Task HandleAsync_GivenOtherOrganisationRepresentsCompanies_IsRepres
+                new DirectProducerSubmission { ComplianceYear = 2024, DirectProducerSubmissionStatus = DirectProducerSubmissionStatus.Incomplete }
+            };
+
+            A.CallTo(() => directRegistrant.DirectProducerSubmissions).Returns(submissions);
+
+            var directRegistrants = new List<DirectRegistrant> { directRegistrant };
+            A.CallTo(() => context.DirectRegistrants).Returns(dbHelper.GetAsyncEnabledDbSet(directRegistrants));
+
+            var message = new GetOrganisationInfo(organisationId);
+
+            // Act
+            var result = await handler.HandleAsync(message);
+
+            // Assert
+            result.DirectRegistrants.Single().MostRecentSubmittedYear.Should().Be(2023);
+        }
+
+        [Fact]
+        public async Task HandleAsync_GivenDirectRegistrantWithMixedStatusSubmissions_ShouldOnlyConsiderCompleteAndReturnedYears()
+        {
+            // Arrange
+            var organisation = GetOrganisationWithId(organisationId);
+            var expectedReturnValue = new OrganisationData();
+            A.CallTo(() => map.Map(A<Organisation>._)).Returns(expectedReturnValue);
+
+            var directRegistrant = A.Fake<DirectRegistrant>();
+            var directRegistrantId = Guid.NewGuid();
+
+            A.CallTo(() => directRegistrant.Id).Returns(directRegistrantId);
+            A.CallTo(() => directRegistrant.OrganisationId).Returns(organisation.Id);
+
+            var submissions = new List<DirectProducerSubmission>
+            {
+                new DirectProducerSubmission { ComplianceYear = 2022, DirectProducerSubmissionStatus = DirectProducerSubmissionStatus.Complete },
+                                new DirectProducerSubmission { ComplianceYear = 2024, DirectProducerSubmissionStatus = DirectProducerSubmissionStatus.Incomplete },
+                new DirectProducerSubmission { ComplianceYear = 2021, DirectProducerSubmissionStatus = DirectProducerSubmissionStatus.Returned }
+            };
+
+            A.CallTo(() => directRegistrant.DirectProducerSubmissions).Returns(submissions);
+
+            var directRegistrants = new List<DirectRegistrant> { directRegistrant };
+            A.CallTo(() => context.DirectRegistrants).Returns(dbHelper.GetAsyncEnabledDbSet(directRegistrants));
+
+            var message = new GetOrganisationInfo(organisationId);
+
+            // Act
+            var result = await handler.HandleAsync(message);
+
+            // Assert
+            result.DirectRegistrants.Single().MostRecentSubmittedYear.Should().Be(2022);
+        }
+
         private Organisation GetOrganisationWithId(Guid id)
         {
             var organisation = A.Fake<Organisation>();
             A.CallTo(() => organisation.Id).Returns(id);
             return organisation;
         }
-
-        private Aatf CreateAatf(Organisation organisation)
```

Comment:
I take it this just wasn't being used anywhere?

Reply:
nope...